examples/main.tf

```diff
@@ -0,0 +1,194 @@
+locals {
+  name = "boldlink-as"
+}
+
+module "complete" {
+  source = "./../"
+
+  ## Autoscaling group
+  name                      = local.name
+  launch_template_name      = local.name
+  min_size                  = 0
+  max_size                  = 1
+  desired_capacity          = 1
+  wait_for_capacity_timeout = 0
+  health_check_type         = "EC2"
+  availability_zones        = ["eu-west-1a", "eu-west-1b", "eu-west-1c"]
+
+  initial_lifecycle_hooks = [
+    {
+      name                  = "LaunchHook"
+      default_result        = "CONTINUE"
+      heartbeat_timeout     = 90
+      lifecycle_transition  = "autoscaling:EC2_INSTANCE_LAUNCHING"
+      notification_metadata = <<EOF
+{
+  "Greetings": "Dev"
+}
+EOF
+    },
+    {
+      name                  = "TermHook"
+      default_result        = "CONTINUE"
+      heartbeat_timeout     = 180
+      lifecycle_transition  = "autoscaling:EC2_INSTANCE_TERMINATING"
+      notification_metadata = <<EOF
+{
+  "Terminating": "Env"
+}
+EOF
+    }
+  ]
+
+  instance_refresh = {
+    strategy = "Rolling"
+    preferences = {
+      checkpoint_delay       = 600
+      checkpoint_percentages = [50, 75, 100]
+      instance_warmup        = 300
+      min_healthy_percentage = 50
+    }
+    triggers = ["tag"]
+  }
+  ## security group: Additional rules
+  security_group_rules = {
+    ingress_http = {
+      from_port   = 80
+      to_port     = 80
+      protocol    = "tcp"
+      cidr_blocks = ["0.0.0.0/0"]
+      type        = "ingress"
+    }
+    custom = {
+      from_port   = 8080
+      to_port     = 8080
+      protocol    = "tcp"
+      cidr_blocks = ["0.0.0.0/0"]
+      type        = "ingress"
+    }
+  }
+  # Launch template
+  launch_template_description = "Complete launch template example"
+  update_default_version      = true
+  create_launch_template      = true
+  image_id                    = data.aws_ami.amazon_linux.id
+  instance_type               = "t3.micro"
+  ebs_optimized               = true
+
+  block_device_mappings = [
+    {
+      # Root volume
+      device_name = "/dev/xvda"
+      no_device   = 0
+      ebs = {
+        delete_on_termination = true
+        volume_size           = 20
+        volume_type           = "gp2"
+      }
+      }, {
+      device_name = "/dev/sda1"
+      no_device   = 1
+      ebs = {
+        delete_on_termination = true
+        volume_size           = 30
+        volume_type           = "gp2"
+      }
+    }
+  ]
+
+  capacity_reservation_specification = {
+    capacity_reservation_preference = "open"
+  }
+
+  cpu_options = {
+    core_count       = 1
+    threads_per_core = 1
+  }
+
+  credit_specification = {
+    cpu_credits = "standard"
+  }
+  /*
+  network_interfaces = [
```

Comment:
It makes more sense to do something like this @patrickmukumbu 
https://github.com/HDE/terraform-aws-autoscaling/blob/master/main.tf#L259-L276

Reply:
Voila!!That works, thanks :smile:  @hugoalmeida264